src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingPartService.java

```diff
@@ -76,8 +77,18 @@ public UserTransaction assignHearingPartToSessionWithTransaction(UUID hearingPar
                                                                      HearingPartSessionRelationship assignment)
                                                                         throws IOException {
         HearingPart hearingPart = hearingPartRepository.findOne(hearingPartId);
+        if(hearingPart.getVersion() != assignment.getHearingPartVersion()) {
+            throw new OptimisticLockException(
+                "HearingPart version mismatch: " + assignment.getHearingPartVersion() + " vs " + hearingPart.getVersion()
+            );
+        }
 
         Session targetSession = sessionRepository.findOne(assignment.getSessionId());
+        if(targetSession.getVersion() != assignment.getSessionVersion()) {
```

Comment:
same as above, this should not be here. it is already implemented elsewhere if detach was used this would not be needed.

Reply:
fixed